File: src/main/java/com/matag/game/turn/action/_continue/ContinueService.java
package com.matag.game.turn.action._continue;

import org.springframework.stereotype.Component;

import com.matag.game.status.GameStatus;
import com.matag.game.turn.phases.PhaseFactory;

import lombok.AllArgsConstructor;

@Component
@AllArgsConstructor
public class ContinueService {
  private final PhaseFactory phaseFactory;

  public void set(GameStatus gameStatus) {
    var phase = phaseFactory.get(gameStatus.getTurn().getCurrentPhase());
    phase.set(gameStatus);
  }

  public void next(GameStatus gameStatus) {
    var phase = phaseFactory.get(gameStatus.getTurn().getCurrentPhase());
    phase.next(gameStatus);
  }
}

File: src/main/kotlin/com/matag/game/turn/action/_continue/ContinueService.kt
package com.matag.game.turn.action._continue

import com.matag.game.status.GameStatus
import com.matag.game.turn.phases.PhaseFactory
import org.springframework.stereotype.Component

@Component
class ContinueService(
    private val phaseFactory: PhaseFactory
) {

    fun set(gameStatus: GameStatus) {
        val phase = phaseFactory.get(gameStatus.turn.currentPhase!!)
        phase.set(gameStatus)
    }

    fun next(gameStatus: GameStatus) {
        val phase = phaseFactory.get(gameStatus.turn.currentPhase!!)
        phase.next(gameStatus)
    }
}
